src/bin/lpython.cpp

```diff
@@ -649,19 +711,27 @@ int compile_to_binary_wasm(
     Allocator al(4*1024);
     LFortran::diag::Diagnostics diagnostics;
     LFortran::LocationManager lm;
-    lm.in_filename = infile;
     std::vector<std::pair<std::string, double>>times;
-    auto file_reading_start = std::chrono::high_resolution_clock::now();
-    std::string input = LFortran::read_file(infile);
-    auto file_reading_end = std::chrono::high_resolution_clock::now();
-    times.push_back(std::make_pair("File reading", std::chrono::duration<double, std::milli>(file_reading_end - file_reading_start).count()));
-    lm.init_simple(input);
+    {
+        LFortran::LocationManager::FileLocations fl;
+        fl.in_filename = infile;
+        lm.files.push_back(fl);
+
+        auto file_reading_start = std::chrono::high_resolution_clock::now();
+        std::string input = LFortran::read_file(infile);
+        auto file_reading_end = std::chrono::high_resolution_clock::now();
+        times.push_back(std::make_pair("File reading", std::chrono::duration
+            <double, std::milli>(file_reading_end - file_reading_start).count()));
+
+        lm.init_simple(input);
+        lm.file_ends.push_back(input.size());
+    }
     auto parsing_start = std::chrono::high_resolution_clock::now();
     LFortran::Result<LFortran::LPython::AST::ast_t*> r = parse_python_file(
-        al, runtime_library_dir, infile, diagnostics, compiler_options.new_parser);
+        al, runtime_library_dir, infile, diagnostics, 0, compiler_options.new_parser);
     auto parsing_end = std::chrono::high_resolution_clock::now();
     times.push_back(std::make_pair("Parsing", std::chrono::duration<double, std::milli>(parsing_end - parsing_start).count()));
-    std::cerr << diagnostics.render(input, lm, compiler_options);
+    // std::cerr << diagnostics.render(lm, compiler_options);
```

Comment:
I think the errors have to be rendered here:
```suggestion
    std::cerr << diagnostics.render(lm, compiler_options);
```